lib/TestPlanBuilder.lua

```diff
@@ -46,6 +48,16 @@ function TestPlanBuilder:getCurrentNode()
 	return self.nodeStack[#self.nodeStack] or self.plan
 end
 
+local function constructFullName(phrase, node)
+	if node then
+		local parent = constructFullName(node.phrase, node.parent)
+		if parent then
+			return parent .. " " .. phrase
+		end
+	end
+	return phrase
+end
```

Comment:
This function is odd!

We could probably rewrite it to just accept a node:

```lua
local function constructFullName(node)
	if node.parent then
		return string.format("%s.%s", constructFullName(node.Parent), node.phrase)
	else
		return node.phrase
	end
end
```

This has the downside that we have to have the node constructed before we can get its name, which affects the addition on line 76. It wouldn't be a big deal to construct the node, check its full name, then mutate the modifier though, since the plan nodes are POD.

Reply:
The best you can get is:
```
local function constructFullName(node)
	if node.parent then
		local parentPhrase = constructFullName(node.parent)
		if parentPhrase then
			return parentPhrase .. " " .. node.phrase
		end
	end
	return node.phrase
end
```
because parent's phrase can be nil as well and you should not have leading dot in name